book/src/input/how_to_define_custom_control_bindings.md

```diff
@@ -0,0 +1,172 @@
+# How to Define Custom Control Bindings
+
+Instead of using `StringBindings` for an `InputBundle` you probably want to use a custom type in production, as `StringBindings` are mainly meant to be used for prototyping and not very efficient.
+
+Using a custom type to handle input instead of using `String` has many advantages:
+
+* A `String` uses quite a lot of memory compared to something like an enum.
+* Inputting a `String` when retrieving input data is error-prone if you mistype it or change the name.
+* You will be able to add additional information to a custom type.
+
+## Defining a Custom Type
+
+Defining a custom type for the `InputBundle` is done by implementing the `BindingTypes` trait. This trait contains two types, an `Axis` type and an `Action` type. These types are usually defined as enums, but you can use whatever you want.
```

Comment:
I'd replace the last sentence with:

These types must implement the following traits: `Clone + Debug + Hash + Eq + Send + Sync + 'static`.
In addition, they must also implement `Serialize` and `Deserialize` to be loaded from a configuration file.

(people newer to Rust don't know what "whatever you want" means).

Reply:
ah, just realized you mention these later on. Can just remove the last sentence then.